serverctl/iptables.go

```diff
@@ -0,0 +1,130 @@
+package serverctl
+
+import (
+	"errors"
+	"net"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/gravitl/netmaker/logger"
+	"github.com/gravitl/netmaker/netclient/ncutils"
+	"github.com/gravitl/netmaker/servercfg"
+)
+
+const NETMAKER_PROCESS_NAME = "netmaker"
+
+// InitServerNetclient - intializes the server netclient
+func InitIPTables() error {
+	_, err := exec.LookPath("iptables")
+	if err != nil {
+		return err
+	}
+	err = setForwardPolicy()
+	if err != nil {
+		logger.Log(0, "error setting iptables forward policy: "+err.Error())
+	}
+
+	err = portForwardServices()
+	if err != nil {
+		return err
+	}
+	if isContainerized() && servercfg.IsHostNetwork() {
+		err = setHostCoreDNSMapping()
+	}
+	return err
+}
+
+// set up port forwarding for services listed in config
+func portForwardServices() error {
+	var err error
+	services := servercfg.GetPortForwardServiceList()
+	if len(services) == 0 || services[0] == "" {
+		return nil
+	}
+	for _, service := range services {
+		switch service {
+		case "mq":
+			err = iptablesPortForward("mq", "1883", "1883", false)
+		case "dns":
+			err = iptablesPortForward("coredns", "53", "53", false)
+		case "ssh":
+			err = iptablesPortForward("127.0.0.1", "22", "22", true)
+		default:
+			params := strings.Split(service, ":")
+			err = iptablesPortForward(params[0], params[1], params[2], true)
+		}
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// determine if process is running in container
+func isContainerized() bool {
+	fileBytes, err := os.ReadFile("/proc/1/sched")
+	if err != nil {
+		logger.Log(1, "error determining containerization: "+err.Error())
+		return false
+	}
+	fileString := string(fileBytes)
+	return strings.Contains(fileString, NETMAKER_PROCESS_NAME)
+}
+
+// make sure host allows forwarding
+func setForwardPolicy() error {
+	logger.Log(1, "setting iptables forward policy")
+	_, err := ncutils.RunCmd("iptables --policy FORWARD ACCEPT", false)
+	return err
+}
+
+// port forward from an entry, can contain a dns name for lookup
+func iptablesPortForward(entry string, inport string, outport string, isIP bool) error {
+	logger.Log(1, "forwarding "+entry+" traffic from host port "+inport+" to container port "+outport)
```

Comment:
concatenation not required on logger.Log(1, "blah","blah","blah") will yield blah blah blah